freva-client/src/freva_client/query.py

```diff
@@ -757,3 +830,51 @@ def _get(
                 raise ValueError(msg) from None
             logger.warning(msg)
         return None
+
+    def _put(
+        self, url: str, data: Dict[str, Any], **kwargs: Any
+    ) -> Optional[requests.models.Response]:
+        """Apply the PUT method to the databrowser."""
+        logger.debug(
+            "PUT request to %s with data: %s and parameters: %s",
+            url,
+            data,
+            self._params,
+        )
+        kwargs.setdefault("timeout", 30)
+        params = kwargs.pop("params", {})
+        try:
+            res = requests.put(
+                url, json=data, params={**self._params, **params}, **kwargs
+            )
+            res.raise_for_status()
+            return res
+        except KeyboardInterrupt:
+            pprint("[red][b]User interrupt: Exit[/red][/b]", file=sys.stderr)
+
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.HTTPError,
+        ) as error:
+            msg = f"adding user data request failed with {error}"
+            logger.warning(msg)
+        return None
+
+    def _delete(self, url: str, **kwargs: Any) -> Optional[requests.models.Response]:
+        """Apply the DELETE method to the databrowser."""
+        logger.debug("DELETE request to %s with parameters: %s", url, self._params)
+        params = kwargs.pop("params", {})
+        kwargs.setdefault("timeout", 30)
+        try:
+            res = requests.delete(url, params={**self._params, **params}, **kwargs)
+            res.raise_for_status()
+            return res
+        except KeyboardInterrupt:
+            pprint("[red][b]User interrupt: Exit[/red][/b]", file=sys.stderr)
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.HTTPError,
+        ) as error:
+            msg = f"DELETE request failed with {error}"
+            logger.warning(msg)
```

Comment:
Same for the error here. The get method is in my opinion a little different cause a user can decide on what happens if a search has no results. But things might be a little different in delete or post scenarios. Maybe you could return the number of deleted/added objects. The users can decide on what to do if it's 0 but should be more than 0.